### PR TITLE
transport: move to internal to make room for new, public transport API

### PR DESCRIPTION
**Pull request by dfawley**
This is a breaking change, but the transport package was never intended for use outside of grpc.  Any current users that we are aware of are incorrect or have a preferred alternative.

### CODE REVIEW

File: internal/transport/transport.go [hunk 1]
```diff
@@ -19,7 +19,7 @@
 // Package transport defines and implements message oriented communication
 // channel to complete various transactions (e.g., an RPC).  It is meant for
 // grpc-internal usage and is not intended to be imported directly by users.
-package transport // externally used as import "google.golang.org/grpc/transport"
+package transport // externally used as import "google.golang.org/grpc/internal/transport"
```
Comment (menghanl) [inline internal/transport/transport.go]: Should we just delete the comment on this line?

**Comment by MaerF0x0** [comment]
I forget the name of the laws of APIs, but it goes something like this. Given enough consumers of an API even if not documented/supported  someone is depending on it.  That was me for the internal/transport 

something like: 

```go
			if err == transport.ErrConnClosing {
				log.Printf("[ERROR err=%q", err)
				break SomeJump // The transport is closing, afaik this is not a bug
			}
```

we'll pin to 1.13

**Comment by menghanl** [comment]
@MaerF0x0 `transport.ErrConnClosing` is never returned by gRPC package, it will be wrapped into a [status](https://github.com/grpc/grpc-go/blob/07ef407d991f1004e6c3367c8f452ed9a02f17ff/status/status.go#L28) error. So if you were expecting this error from an RPC, you can safely delete this `if` condition now.

**Comment by damianoneill** [comment]
agree with @MaerF0x0 this is major version increment.

**Comment by nehbit** [comment]
This is major, breaking change and should marked as such, since up until this point this package was externally accessible.

As of now, using this package is the only way to get the remote IP address through GRPC as far as my research yielded, which is a fairly run of the mill need. (With [PeerFromContext](https://github.com/grpc/grpc-go/blob/c9eccd89e06a7c59b5c225abf98b33d384ccaa41/transport/transport.go#L318)). 

The functionality in question is relatively trivial, but this took me a little bit of head scratching this morning.

**Pull request merged**